File: packages/java/src/main/java/com/tesote/sdk/models/Whoami.java
package com.tesote.sdk.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Whoami(@JsonProperty("client") Client client) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Client(
            @JsonProperty("id") String id,
            @JsonProperty("name") String name,
            @JsonProperty("type") String type
    ) {}
}

File: packages/php/src/Models/WhoAmI.php
<?php

declare(strict_types=1);

namespace Tesote\Sdk\Models;

/** Response from GET /whoami and GET /v2/whoami. */
final readonly class WhoAmI
{
    public function __construct(
        public string $id,
        public string $name,
        public string $type,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        $client = is_array($data['client'] ?? null) ? $data['client'] : [];
        return new self(
            id: (string) ($client['id'] ?? ''),
            name: (string) ($client['name'] ?? ''),
            type: (string) ($client['type'] ?? ''),
        );
    }
}
